locations/jclouds/src/main/java/org/apache/brooklyn/location/jclouds/networking/JcloudsLocationSecurityGroupCustomizer.java

```diff
@@ -223,56 +230,64 @@ private SecurityGroup getSecurityGroup(final String nodeId, final SecurityGroupE
      * @param location Location to gain permissions
      * @param permissions The set of permissions to be applied to the location
      */
-    public JcloudsLocationSecurityGroupCustomizer addPermissionsToLocation(final JcloudsMachineLocation location, final Iterable<IpPermission> permissions) {
-        ComputeService computeService = location.getParent().getComputeService();
-        addPermissionsToLocationAndReturnSecurityGroup(computeService, location, permissions);
+    public JcloudsLocationSecurityGroupCustomizer addPermissionsToLocation(final JcloudsMachineLocation location,
+              final Iterable<IpPermission> permissions) {
+        addPermissionsToLocationAndReturnSecurityGroup(location, permissions);
         return this;
     }
 
-    public Collection<SecurityGroup> addPermissionsToLocationAndReturnSecurityGroup(ComputeService computeService, final JcloudsMachineLocation location, final Iterable<IpPermission> permissions) {
+    public Collection<SecurityGroup> addPermissionsToLocationAndReturnSecurityGroup(
+        final JcloudsMachineLocation location, final Iterable<IpPermission> permissions) {
+
         synchronized (JcloudsLocationSecurityGroupCustomizer.class) {
```

Comment:
I'm asking because of [this](https://github.com/apache/brooklyn-server/blob/master/locations/jclouds/src/main/java/org/apache/brooklyn/location/jclouds/networking/JcloudsLocationSecurityGroupCustomizer.java#L217-L221) comment. Looks like the same restrictions will apply on the synchronizer.
Agree it can be tackled in a separate PR.